### ISSUE TITLE
False positive for go.lang.security.audit.net.pprof

### ISSUE THREAD

**Issue by dennwc** [returntocorp/sgrep-rules#1537]
**Describe the bug**

The rule `go.lang.security.audit.net.pprof` assumes that any calls to `http.ListenAndServe` are insecure if the `_ "net/http/pprof"` import is present. However in practice it's not always the case.

The `net/http/pprof` package only registers handlers on `http.DefaultServeMux`, thus making all `http.ListenAndServe(..., nil)` calls potentially insecure. The caller can override the handler by passing a secord argument instead of `nil`. Currently `semgrep` reports in issue even when "clean" HTTP handler is passed to the function.  

**To Reproduce**

```go
mux := http.NewServeMux()
log.Fatal(http.ListenAndServe(":8080", mux))
```

**Expected behavior**

The code above should produce no warning, because even though the HTTP server is exposed, it serves a new empty `ServeMux` instead of `http.DefaultServeMux` (which is "poisoned" by pprof).

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [x] P2: annoying but not blocking me

**Comment by minusworld** [comment]
@dennwc Thanks! Your explanation was excellent. 😄 

If you have a chance, does https://github.com/returntocorp/semgrep-rules/pull/1544 look like it addresses this issue?

**Comment by dennwc** [comment]
Haven't tested the PR, but looking at the change it will solve _our_ issue, but may still fail for other cases, unfortunately.

The handler passed to that function can be anything else other than `http.DefaultMux` or `nil`, but that would probably be hard to express, right?

I think "anything not `nil`" would be a safe bet and will work for most cases. The only false _negative_ would be if someone explicitly passes `http.DefaultMux` (possible via variable).
